internal/jobs/launch_instance_aws.go

```diff
@@ -2,60 +2,162 @@ package jobs
 
 import (
 	"context"
+	"errors"
 	"fmt"
 
 	"github.com/RHEnVision/provisioning-backend/internal/clients"
+	"github.com/RHEnVision/provisioning-backend/internal/clients/http"
 	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
 	"github.com/RHEnVision/provisioning-backend/internal/dao"
 	"github.com/RHEnVision/provisioning-backend/internal/models"
 	"github.com/RHEnVision/provisioning-backend/internal/userdata"
+	"github.com/RHEnVision/provisioning-backend/pkg/worker"
 	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
-	"github.com/lzap/dejq"
 )
 
 type LaunchInstanceAWSTaskArgs struct {
 	// Associated reservation
-	ReservationID int64 `json:"reservation_id"`
+	ReservationID int64
 
 	// Associated account
-	AccountID int64 `json:"account_id"`
+	AccountID int64
 
 	// Region to provision the instances into
-	Region string `json:"region"`
+	Region string
 
 	// Associated public key
-	PubkeyID int64 `json:"pubkey_id"`
+	PubkeyID int64
 
 	// Source ID that was used to get the ARN
 	SourceID string
 
 	// Detail information
-	Detail *models.AWSDetail `json:"detail"`
+	Detail *models.AWSDetail
 
 	// AWS AMI as fetched from image builder
-	AMI string `json:"ami"`
+	AMI string
 
 	// The ARN fetched from Sources which is linked to a specific source
-	ARN *clients.Authentication `json:"arn"`
+	ARN *clients.Authentication
 }
 
 // Unmarshall arguments and handle error
-func HandleLaunchInstanceAWS(ctx context.Context, job dejq.Job) error {
-	args := LaunchInstanceAWSTaskArgs{}
-	err := decodeJob(ctx, job, &args)
-	if err != nil {
-		return err
+func HandleLaunchInstanceAWS(ctx context.Context, job *worker.Job) {
+	args, ok := job.Args.(LaunchInstanceAWSTaskArgs)
+	if !ok {
+		ctxval.Logger(ctx).Error().Msgf("Type assertion error for job %s, unable to finish reservation: %#v", job.ID, job.Args)
+		return
+	}
+
+	ctx = contextLogger(ctx, args.AccountID, args.ReservationID)
+
+	jobErr := DoEnsurePubkeyOnAWS(ctx, &args)
+	if jobErr != nil {
+		finishWithError(ctx, args.ReservationID, jobErr)
+		return
 	}
-	ctx = contextLogger(ctx, job.Type(), args, args.AccountID, args.ReservationID)
 
-	jobErr := handleLaunchInstanceAWS(ctx, &args)
+	jobErr = DoLaunchInstanceAWS(ctx, &args)
 
 	finishJob(ctx, args.ReservationID, jobErr)
-	return jobErr
 }
 
 // Job logic, when error is returned the job status is updated accordingly
-func handleLaunchInstanceAWS(ctx context.Context, args *LaunchInstanceAWSTaskArgs) error {
+func DoEnsurePubkeyOnAWS(ctx context.Context, args *LaunchInstanceAWSTaskArgs) error {
```

Comment:
All functions ("steps") now start with `DoXXX` and are exported so we can write tests in a separate `jobs_test` package (no cycles).